### ISSUE TITLE
Fastlane gym command blocked by [CP] Prepare Artifacts phase

### ISSUE THREAD

**Issue by Narayane** [CocoaPods/CocoaPods#9857]
* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

Run a Fastlane deployment lane

## What did you expect to happen?

Deploy an ipa on AppCenter platform

## What happened instead?

Fastlane `gym` command is stuck by a new Cocoapods script added in my Xcode build phases. 

```
[CP] Prepare Artifacts
/bin/sh "${PODS_ROOT}/Target Support Files/Pods-xxx/Pods-xxx-artifacts.sh"
```

I have posted [this issue](https://github.com/fastlane/fastlane/issues/16590) on Fastlane but I'm pretty sure now that my problem is a side effect of the adding of `[CP] Prepare Artifacts` phase because it is the only difference between a running and stuck lane state.

### Stack

```
   CocoaPods : 1.9.3
        Ruby : ruby 2.6.3p62 (2019-04-16 revision 67580) [universal.x86_64-darwin19]
    RubyGems : 3.0.3
        Host : Mac OS X 10.15.5 (19F101)
       Xcode : 11.5 (11E608c)
         Git : git version 2.24.3 (Apple Git-128)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib
Repositories : cocoapods - git - https://github.com/CocoaPods/Specs.git @ 9291339b65a4e4ea738e6d6f4242c75df06cb6f1

```

### Installation Source

```
Executable Path: /usr/local/bin/pod
```

### Plugins

```
cocoapods-check       : 1.1.0
cocoapods-deintegrate : 1.0.4
cocoapods-keys        : 2.2.0
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.1.0
cocoapods-trunk       : 1.5.0
cocoapods-try         : 1.2.0
```

### Podfile

```ruby
platform :ios, '12.0'
# ignore all warnings from all pods
inhibit_all_warnings!
use_frameworks!

# Pod repositories
source 'https://github.com/CocoaPods/Specs.git' # default

# versions
$firebase_version = '~> 6.5'
$facebook_version = '~> 7.0'
$rxswift_version = '~> 5.0'

# logical pods groups
def pods_analytics
  pod 'Fabric'
  pod 'Crashlytics'
  pod 'Amplitude-iOS', '~> 4.9'
  pod 'Batch', '~> 1.14'
  pod 'AppsFlyerFramework', '~> 5.0'
end

def pods_google_utilities
  pod 'GoogleUtilities/AppDelegateSwizzler'
  pod 'GoogleUtilities/Environment'
  pod 'GoogleUtilities/ISASwizzler'
  pod 'GoogleUtilities/Logger'
  pod 'GoogleUtilities/MethodSwizzler'
  pod 'GoogleUtilities/NSData+zlib'
  pod 'GoogleUtilities/Network'
  pod 'GoogleUtilities/Reachability'
  pod 'GoogleUtilities/UserDefaults'
  pod 'GTMSessionFetcher/Core'
end

def pods_firebase
  pod 'Firebase/Core', $firebase_version
  pod 'Firebase/Storage', $firebase_version 
  pod 'Firebase/Database', $firebase_version
  pod 'Firebase/Auth', $firebase_version
  pod 'Firebase/RemoteConfig', $firebase_version
  pod 'Firebase/Messaging', $firebase_version
  pod 'Firebase/DynamicLinks', $firebase_version
  pod 'Firebase/Performance', $firebase_version
end

def pods_social
  pod 'GoogleSignIn', '~> 5.0'
  pod 'FBSDKCoreKit', $facebook_version
  pod 'FBSDKLoginKit', $facebook_version
  pod 'FBSDKShareKit', $facebook_version
end

def pods_rx
  pod 'RxSwift', $rxswift_version
  pod 'RxCocoa', $rxswift_version
  pod 'RxGesture', '~> 3.0'
  pod 'RxFirebaseAuth', '~> 2.4'
  pod 'RxMKMapView', $rxswift_version
  pod 'RxCoreLocation', '~> 1.4'
  pod 'RxMediaPicker', :git => 'https://github.com/liberty-rider/RxMediaPicker.git'
  pod 'RxOptional', '~> 4.0'
  pod 'RxSwiftExt', $rxswift_version
  pod 'RxDataSources', '~> 4.0.1'
end

def pods_di
  pod 'Dip', '~> 7.1'
end

def pods_geometry
  pod 'Polyline', '~> 4.2'
  pod 'GeoFeatures', '~> 1.7'
  pod 'SwiftTurf', '~> 0.3'
  pod 'Turf', '~> 0.3'
end

def pods_widgets
  pod 'Cosmos', '~> 22.0'
  pod 'MZTimerLabel', '0.5.4'
  pod 'NVActivityIndicatorView', '~> 4.3'
  pod 'PKHUD', '~> 5.1'
  pod 'ChromaColorPicker', '~> 2.0'
  pod "FlagPhoneNumber", '~> 0.8'
end

def pods_helpers
  pod 'SwiftyBeaver', '~> 1.6'
  pod 'PhoneNumberKit', '~> 3.0'
  pod 'SwipeCellKit', '~> 2.6'
  pod 'AlamofireImage', '~> 4.0'
  pod 'IQKeyboardManagerSwift', '~> 6.1'
  pod 'Connectivity', '~> 4.0'
  pod 'JWTDecode', '~> 2.1'
end

def pods_math
  pod 'Surge', '~> 2.3.0'
  pod 'SigmaSwiftStatistics', '~> 9.0'
end

# test logical pods groups
def pods_tests_rx
  pod 'RxSwift', $rxswift_version
  pod 'RxBlocking', $rxswift_version
  pod 'RxTest', $rxswift_version
end

# target
target 'xxx' do
  pods_google_utilities
  pods_analytics
  pods_firebase
  pods_social
  pods_rx
  pods_di
  pods_widgets
  pods_helpers
  pods_geometry
  pods_math
  pod 'Apollo/SQLite', '~> 0.27'
  pod 'SwiftQueue', '~> 4.0'
  pod 'HEREMaps', '~> 3.15'
  pod 'Purchases', '~> 3.0'
  pod 'lottie-ios', '~> 3.1'
  pod 'ViewAnimator', '~> 2.7'

 post_install do |installer|
    installer.pods_project.targets.each do |target|
      target.build_configurations.each do |config|
        config.build_settings['LD_NO_PIE'] = 'NO'
        config.build_settings['CLANG_ANALYZER_LOCALIZABILITY_NONLOCALIZED'] = 'YES'
        if target.name == 'Mockit'
          config.build_settings['ENABLE_BITCODE'] = 'NO'
        end
        if config.name == 'Release'
          config.build_settings['SWIFT_OPTIMIZATION_LEVEL'] = '-Owholemodule' # remove default warning on Xcode
        else
          config.build_settings['SWIFT_OPTIMIZATION_LEVEL'] = '-Onone'
        end
        config.build_settings['CLANG_ENABLE_CODE_COVERAGE'] = 'NO' # exclude pods of code coverage
      end
    end
  end
end
```
Can anyone help me please?

**Comment by joaopdcgarcia** [comment]
Its happening the same to me!

**Comment by anoopnimkar** [comment]
Same here! experiencing the same issue - getting stuck at `Running script [CP] Prepare Artifacts` and build times out

**Comment by amorde** [comment]
Do you see any output in the build log?

FWIW this script will be removed in the next version, but happy to help troubleshoot. I don't think the Podfile alone will help here, would need either an example project or some additional details on what it is getting stuck on

**Comment by anoopnimkar** [comment]
No output on the build log, it just gets stuck on `Prepare Artifcats` - using Circle CI here. Will try to get the example project. Same works locally but not on CI

`[07:23:05]: ▸ Signing /Users/distiller/Library/Developer/Xcode/DerivedData/Project-hhvuwikeeyrxgmfwontmvbaxpqkf/Build/Products/Debug-iphonesimulator/Pods_Tours.framework
 
[07:23:06]: ▸ Building Project/Tours [Debug]
 
[07:23:06]: ▸ Check Dependencies
 
[07:23:06]: ▸ Running script '[CP] Check Pods Manifest.lock'
 
[07:23:06]: ▸ Running script '[CP] Prepare Artifacts'
 
 
Too long with no output (exceeded 1h30m0s): context deadline exceeded 
`

**Comment by amorde** [comment]
there's extra output that isn't shown there, can you enable any verbose flags within Gym/Fastlane?

**Comment by Narayane** [comment]
Hi @amorde,

I added `--verbose` to my `bundle exec fastlane ios myLane` command but I don't think it will be very helpful... What is interesting is Fastlane `scan` command run just before `gym` in my lane myLane is not impacted by `Running script '[CP] Prepare Artifacts'` when it called...

```
INFO [2020-06-17 23:12:33.00]: -----------------
INFO [2020-06-17 23:12:33.00]: --- Step: gym ---
INFO [2020-06-17 23:12:33.00]: -----------------
DEBUG [2020-06-17 23:12:33.00]: Merging provisioning profile mappings
DEBUG [2020-06-17 23:12:33.00]: -------------------------------------
DEBUG [2020-06-17 23:12:33.00]: Primary provisioning profile mapping:
DEBUG [2020-06-17 23:12:33.00]: {}
DEBUG [2020-06-17 23:12:33.00]: Secondary provisioning profile mapping:
DEBUG [2020-06-17 23:12:33.00]: {:"xxx"=>"match Development xxx", :"xxx.watchkitapp"=>"match Development xxx.watchkitapp", :"xxx.watchkitapp.watchkitextension"=>"match Development xxx.watchkitapp.watchkitextension"}
DEBUG [2020-06-17 23:12:33.00]: Resulting in the following mapping:
DEBUG [2020-06-17 23:12:33.00]: {:"xxx"=>"match Development xxx", :"xxx.watchkitapp"=>"match Development xxx.watchkitapp", :"xxx.watchkitapp.watchkitextension"=>"match Development xxx.watchkitapp.watchkitextension"}
WARN [2020-06-17 23:12:33.01]: Resolving Swift Package Manager dependencies...
INFO [2020-06-17 23:12:33.01]: $ xcodebuild -resolvePackageDependencies -workspace xxx.xcworkspace -scheme Debug -configuration Debug
INFO [2020-06-17 23:12:33.79]: ▸ Command line invocation:
INFO [2020-06-17 23:12:33.79]: ▸     /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild -resolvePackageDependencies -workspace "xxx.xcworkspace" -scheme Debug -configuration Debug
INFO [2020-06-17 23:12:35.35]: ▸ resolved source packages: 
INFO [2020-06-17 23:12:35.36]: $ xcodebuild -showBuildSettings -workspace xxx.xcworkspace -scheme Debug -configuration Debug
WARN [2020-06-17 23:12:38.36]: Command timed out after 3 seconds on try 1 of 4, trying again with a 6 second timeout...
DEBUG [2020-06-17 23:12:41.31]: Parsing project file '/Users/me/Workspace/xxx/angel-ios/xxx.xcodeproj' to find selected provisioning profiles
DEBUG [2020-06-17 23:12:41.31]: Finding provision profiles for 'Debug'
DEBUG [2020-06-17 23:12:41.49]: Merging provisioning profile mappings
DEBUG [2020-06-17 23:12:41.49]: -------------------------------------
DEBUG [2020-06-17 23:12:41.49]: Primary provisioning profile mapping:
DEBUG [2020-06-17 23:12:41.49]: {:"xxx"=>"match Development xxx", :"xxx.watchkitapp"=>"match Development xxx.watchkitapp", :"xxx.watchkitapp.watchkitextension"=>"match Development xxx.watchkitapp.watchkitextension"}
DEBUG [2020-06-17 23:12:41.49]: Secondary provisioning profile mapping:
DEBUG [2020-06-17 23:12:41.49]: {:"xxx"=>"match Development xxx", :"xxx.watchkitapp"=>"match Development xxx.watchkitapp", :"xxx.watchkitapp.watchkitextension"=>"match Development xxx.watchkitapp.watchkitextension"}
DEBUG [2020-06-17 23:12:41.49]: Resulting in the following mapping:
DEBUG [2020-06-17 23:12:41.49]: {:"xxx"=>"match Development xxx", :"xxx.watchkitapp"=>"match Development xxx.watchkitapp", :"xxx.watchkitapp.watchkitextension"=>"match Development xxx.watchkitapp.watchkitextension"}
INFO [2020-06-17 23:12:41.50]: Detected provisioning profile mapping: {:"xxx"=>"match Development xxx", :"xxx.watchkitapp"=>"match Development xxx.watchkitapp", :"xxx.watchkitapp.watchkitextension"=>"match Development xxx.watchkitapp.watchkitextension"}

+-----------------------------------------------------------------------------------------------+---------------------------------------------------------------------------------+
|                                                                             Summary for gym 2.149.1                                                                             |
+-----------------------------------------------------------------------------------------------+---------------------------------------------------------------------------------+
| workspace                                                                                     | xxx.xcworkspace                                                       |
| scheme                                                                                        | Debug                                                                           |
| configuration                                                                                 | Debug                                                                           |
| export_method                                                                                 | development                                                                     |
| output_directory                                                                              | ./artifacts/                                                                    |
| output_name                                                                                   | xxx                                                                    |
| xcargs                                                                                        | -UseModernBuildSystem=NO                                                        |
| export_options.provisioningProfiles.xxx                           | match Development xxx                               |
| export_options.provisioningProfiles.xxx.watchkitapp               | match Development xxx.watchkitapp                   |
| export_options.provisioningProfiles.xxx.watchkitapp.watchkitexte  | match Development xxx.watchkitapp.watchkitextension |
| nsion                                                                                         |                                                                                 |
| destination                                                                                   | generic/platform=iOS                                                            |
| build_path                                                                                    | /Users/me/Library/Developer/Xcode/Archives/2020-06-17              |
| clean                                                                                         | false                                                                           |
| silent                                                                                        | false                                                                           |
| skip_package_ipa                                                                              | false                                                                           |
| skip_package_pkg                                                                              | false                                                                           |
| result_bundle                                                                                 | false                                                                           |
| buildlog_path                                                                                 | ~/Library/Logs/gym                                                              |
| skip_profile_detection                                                                        | false                                                                           |
| xcode_path                                                                                    | /Applications/Xcode.app                                                         |
+-----------------------------------------------------------------------------------------------+---------------------------------------------------------------------------------+

+-----------------------------------------------------------------------------------------------+----------------------+
|                                               Generated Build Command                                                |
+-----------------------------------------------------------------------------------------------+----------------------+
| Option                                                                                        | Value                |
+-----------------------------------------------------------------------------------------------+----------------------+
| set -o pipefail &&                                                                            |                      |
| xcodebuild                                                                                    |                      |
| -workspace xxx.xcworkspace                                                         |                      |
| -scheme Debug                                                                                 |                      |
| -configuration Debug                                                                          |                      |
| -destination                                                                                  | generic/platform=iOS |
| -archivePath                                                                                  |                      |
| /Users/me/Library/Developer/Xcode/Archives/2020-06-17/xxx\ 2020-06-17\  |                      |
| 23.12.41.xcarchive                                                                            |                      |
| -UseModernBuildSystem=NO                                                                      |                      |
| archive                                                                                       |                      |
| | tee /Users/me/Library/Logs/gym/xxx-Debug.log                        |                      |
| | xcpretty                                                                                    |                      |
+-----------------------------------------------------------------------------------------------+----------------------+
INFO [2020-06-17 23:12:41.61]: $ set -o pipefail && xcodebuild -workspace xxx.xcworkspace -scheme Debug -configuration Debug -destination 'generic/platform=iOS' -archivePath /Users/me/Library/Developer/Xcode/Archives/2020-06-17/xxx\ 2020-06-17\ 23.12.41.xcarchive -UseModernBuildSystem=NO archive | tee /Users/me/Library/Logs/gym/xxx-Debug.log | xcpretty
INFO [2020-06-17 23:12:45.93]: ▸ Building Pods/Amplitude-iOS [Debug]
INFO [2020-06-17 23:12:45.93]: ▸ Check Dependencies
INFO [2020-06-17 23:12:45.96]: ▸ Processing Amplitude-iOS-Info.plist
INFO [2020-06-17 23:12:46.01]: ▸ Copying AMPConstants.h
INFO [2020-06-17 23:12:46.01]: ▸ Copying AMPDatabaseHelper.h
INFO [2020-06-17 23:12:46.01]: ▸ Copying AMPDeviceInfo.h
INFO [2020-06-17 23:12:46.01]: ▸ Copying AMPIdentify.h
INFO [2020-06-17 23:12:46.01]: ▸ Copying Amplitude+SSLPinning.h
INFO [2020-06-17 23:12:46.01]: ▸ Copying Amplitude-iOS-umbrella.h
INFO [2020-06-17 23:12:46.01]: ▸ Copying Amplitude.h
INFO [2020-06-17 23:12:46.01]: ▸ Copying AMPLocationManagerDelegate.h
INFO [2020-06-17 23:12:46.01]: ▸ Copying AMPRevenue.h
INFO [2020-06-17 23:12:46.01]: ▸ Copying AMPTrackingOptions.h
INFO [2020-06-17 23:12:46.02]: ▸ Copying AMPURLConnection.h
INFO [2020-06-17 23:12:46.02]: ▸ Copying AMPURLSession.h
INFO [2020-06-17 23:12:46.02]: ▸ Copying AMPUtils.h
INFO [2020-06-17 23:12:46.02]: ▸ Copying ISPCertificatePinning.h
INFO [2020-06-17 23:12:46.02]: ▸ Copying ISPPinnedNSURLConnectionDelegate.h
INFO [2020-06-17 23:12:46.02]: ▸ Copying ISPPinnedNSURLSessionDelegate.h
INFO [2020-06-17 23:12:46.03]: ▸ Compiling AMPConstants.m
INFO [2020-06-17 23:12:59.74]: ▸ Compiling AMPDatabaseHelper.m
INFO [2020-06-17 23:12:59.74]: ▸ Compiling AMPDeviceInfo.m
INFO [2020-06-17 23:12:59.74]: ▸ Compiling AMPIdentify.m
INFO [2020-06-17 23:12:59.74]: ▸ Compiling Amplitude-iOS-dummy.m
INFO [2020-06-17 23:12:59.74]: ▸ Compiling Amplitude.m
INFO [2020-06-17 23:12:59.75]: ▸ Compiling AMPLocationManagerDelegate.m
INFO [2020-06-17 23:12:59.75]: ▸ Compiling AMPRevenue.m
INFO [2020-06-17 23:12:59.75]: ▸ Compiling AMPTrackingOptions.m
INFO [2020-06-17 23:12:59.75]: ▸ Compiling AMPURLConnection.m
INFO [2020-06-17 23:12:59.75]: ▸ Compiling AMPURLSession.m
INFO [2020-06-17 23:12:59.75]: ▸ Compiling AMPUtils.m
INFO [2020-06-17 23:12:59.75]: ▸ Compiling ISPCertificatePinning.m
INFO [2020-06-17 23:12:59.75]: ▸ Compiling ISPPinnedNSURLConnectionDelegate.m
INFO [2020-06-17 23:12:59.75]: ▸ Compiling ISPPinnedNSURLSessionDelegate.m
INFO [2020-06-17 23:12:59.75]: ▸ Compiling Amplitude_iOS_vers.c
INFO [2020-06-17 23:12:59.75]: ▸ Compiling AMPConstants.m
INFO [2020-06-17 23:12:59.75]: ▸ Compiling AMPDatabaseHelper.m
INFO [2020-06-17 23:12:59.75]: ▸ Compiling AMPDeviceInfo.m
INFO [2020-06-17 23:12:59.75]: ▸ Compiling AMPIdentify.m
INFO [2020-06-17 23:12:59.76]: ▸ Compiling Amplitude-iOS-dummy.m
INFO [2020-06-17 23:12:59.76]: ▸ Compiling Amplitude.m
INFO [2020-06-17 23:12:59.76]: ▸ Compiling AMPLocationManagerDelegate.m
INFO [2020-06-17 23:12:59.76]: ▸ Compiling AMPRevenue.m
INFO [2020-06-17 23:12:59.76]: ▸ Compiling AMPTrackingOptions.m
INFO [2020-06-17 23:12:59.76]: ▸ Compiling AMPURLConnection.m
INFO [2020-06-17 23:12:59.76]: ▸ Compiling AMPURLSession.m
INFO [2020-06-17 23:12:59.76]: ▸ Compiling AMPUtils.m
INFO [2020-06-17 23:12:59.76]: ▸ Compiling ISPCertificatePinning.m
INFO [2020-06-17 23:12:59.76]: ▸ Compiling ISPPinnedNSURLConnectionDelegate.m
INFO [2020-06-17 23:12:59.76]: ▸ Compiling ISPPinnedNSURLSessionDelegate.m
INFO [2020-06-17 23:12:59.76]: ▸ Compiling Amplitude_iOS_vers.c
INFO [2020-06-17 23:12:59.97]: ▸ Linking Amplitude_iOS
INFO [2020-06-17 23:13:00.41]: ▸ ⚠️  ld: Invalid debug info found, debug info will be stripped
INFO [2020-06-17 23:13:00.41]: ▸ Linking Amplitude_iOS
INFO [2020-06-17 23:13:00.51]: ▸ ⚠️  ld: Invalid debug info found, debug info will be stripped
INFO [2020-06-17 23:13:00.89]: ▸ Copying Amplitude-iOS/Sources/Amplitude/api.amplitude.com.der
INFO [2020-06-17 23:13:00.95]: ▸ Copying Amplitude-iOS/Sources/Amplitude/ComodoCaLimitedRsaCertificationAuthority.der
INFO [2020-06-17 23:13:01.09]: ▸ Copying Amplitude-iOS/Sources/Amplitude/ComodoRsaCA.der
INFO [2020-06-17 23:13:01.13]: ▸ Copying Amplitude-iOS/Sources/Amplitude/ComodoRsaDomainValidationCA.der
INFO [2020-06-17 23:13:01.86]: ▸ Touching Amplitude_iOS.framework
INFO [2020-06-17 23:13:02.45]: ▸ Aggregate Pods/Batch [Debug]
INFO [2020-06-17 23:13:02.45]: ▸ Check Dependencies
... copy and compile a lot of files ...
INFO [2020-06-17 23:15:00.49]: ▸ Touching FirebaseMessaging.framework
INFO [2020-06-17 23:15:00.51]: ▸ Aggregate Pods/Firebase [Debug]
INFO [2020-06-17 23:15:00.51]: ▸ Check Dependencies
INFO [2020-06-17 23:15:00.51]: ▸ Building Pods/RxFirebaseAuth [Debug]
INFO [2020-06-17 23:15:00.51]: ▸ Check Dependencies
INFO [2020-06-17 23:15:00.52]: ▸ Processing RxFirebaseAuth-Info.plist
INFO [2020-06-17 23:15:00.52]: ▸ Copying RxFirebaseAuth-umbrella.h
INFO [2020-06-17 23:15:00.54]: ▸ Compiling Auth+Rx.swift
INFO [2020-06-17 23:15:00.98]: ▸ Compiling Helper.swift
INFO [2020-06-17 23:15:00.98]: ▸ Compiling Result.swift
INFO [2020-06-17 23:15:00.98]: ▸ Compiling User+Rx.swift
INFO [2020-06-17 23:15:01.11]: ▸ Compiling Auth+Rx.swift
INFO [2020-06-17 23:15:01.12]: ▸ Compiling Helper.swift
INFO [2020-06-17 23:15:01.12]: ▸ Compiling Result.swift
INFO [2020-06-17 23:15:01.12]: ▸ Compiling User+Rx.swift
INFO [2020-06-17 23:15:01.13]: ▸ Compiling RxFirebaseAuth-dummy.m
INFO [2020-06-17 23:15:01.24]: ▸ Compiling RxFirebaseAuth_vers.c
INFO [2020-06-17 23:15:01.24]: ▸ Compiling RxFirebaseAuth-dummy.m
INFO [2020-06-17 23:15:01.24]: ▸ Compiling RxFirebaseAuth_vers.c
INFO [2020-06-17 23:15:01.26]: ▸ Touching RxFirebaseAuth.framework
INFO [2020-06-17 23:15:01.28]: ▸ Building Pods/Pods-xxx [Debug]
INFO [2020-06-17 23:15:01.28]: ▸ Check Dependencies
INFO [2020-06-17 23:15:01.28]: ▸ Processing Pods-xxx-Info.plist
INFO [2020-06-17 23:15:01.29]: ▸ Copying Pods-xxx-umbrella.h
INFO [2020-06-17 23:15:01.30]: ▸ Compiling Pods-xxx-dummy.m
INFO [2020-06-17 23:15:02.14]: ▸ Compiling Pods_xxx_vers.c
INFO [2020-06-17 23:15:02.15]: ▸ Touching Pods_xxx.framework
INFO [2020-06-17 23:15:02.16]: ▸ Building xxx/xxx [Debug]
INFO [2020-06-17 23:15:02.16]: ▸ Check Dependencies
INFO [2020-06-17 23:15:02.18]: ▸ Running script '[CP] Check Pods Manifest.lock'
INFO [2020-06-17 23:15:02.21]: ▸ Running script 'Generate Apollo GraphQL API'
INFO [2020-06-17 23:15:03.59]: ▸ Running script 'Set git hash'
INFO [2020-06-17 23:15:03.67]: ▸ Running script '[CP] Copy Firebase Configuration'
INFO [2020-06-17 23:15:03.69]: ▸ Running script '[CP] Prepare Artifacts'
```

**Comment by amorde** [comment]
Ok, still not quite what I was looking for. When running script phases in Xcode it shows the output of the phase in the build log.

Without additional info there's not much to go off of. I do not know what could be causing this.

**Comment by amorde** [comment]
In the example you posted there's this in the `xcodebuild` invocation:

```
tee /Users/me/Library/Logs/gym/xxx-Debug.log
```

the output should (hopefully) be there

**Comment by Narayane** [comment]
It's the same thing in the relative file... No more relevant infos in my opinion :)

```
Command line invocation:
    /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild -workspace "xxx.xcworkspace" -scheme Debug -configuration Debug -destination generic/platform=iOS -archivePath "/Users/me/Library/Developer/Xcode/Archives/2020-06-17/xxx 2020-06-17 23.12.41.xcarchive" -UseModernBuildSystem=NO archive

User defaults from command line:
    IDEArchivePathOverride = /Users/me/Library/Developer/Xcode/Archives/2020-06-17/xxx 2020-06-17 23.12.41.xcarchive
    UseModernBuildSystem = NO

Prepare build
note: Using legacy build system
=== BUILD TARGET Amplitude-iOS OF PROJECT Pods WITH CONFIGURATION Debug ===

Check dependencies
```

_... Lots of things without interest ..._

```
PhaseScriptExecution [CP]\ Prepare\ Artifacts /Users/me/Library/Developer/Xcode/DerivedData/xxx-ddejunqweisyqocnsoxzgngmnpnw/Build/Intermediates.noindex/ArchiveIntermediates/Debug/IntermediateBuildFilesPath/xxx.build/Debug-iphoneos/xxx.build/Script-C2B2947C14313F313DF788FA.sh
    cd /Users/me/Workspace/xxx/angel-ios
    /bin/sh -c \"/Users/sebastienbalard/Library/Developer/Xcode/DerivedData/xxx-ddejunqweisyqocnsoxzgngmnpnw/Build/Intermediates.noindex/ArchiveIntermediates/Debug/IntermediateBuildFilesPath/xxx.build/Debug-iphoneos/xxx.build/Script-C2B2947C14313F313DF788FA.sh\"
```

**Comment by amorde** [comment]
Darn. I don’t have any ideas unfortunately

**Comment by Narayane** [comment]
I cross my fingers and hope cocoapods 1.10.0 will solve this problem so... :)

**Comment by hanulyuntera** [comment]
I have same issue.. 😰

**Comment by dnkoutso** [comment]
Have you tried the master branch so we can verify this is an issue in 1.9.3?

**Comment by amorde** [comment]
Changes planned for 1.10 include removing the `Prepare Artifacts` script entirely, so it's very possible this is fixed in the master branch currently. Since I can't reproduce this, will have to rely on you testing this against your project

**Comment by joaopdcgarcia** [comment]
> Changes planned for 1.10 include removing the `Prepare Artifacts` script entirely, so it's very possible this is fixed in the master branch currently. Since I can't reproduce this, will have to rely on you testing this against your project

Any idea on when the 1.10 will be released?

**Comment by nareen2** [comment]
It looks like  "cat > $ARTIFACT_LIST_FILE" line in "${PODS_ROOT}/Target Support Files/Pods-{app name}/Pods-{app name}-artifacts.sh" causing this issue as stated in https://sugeul.github.io/2020-06-02-xcodebuild-hang-on-CP-Prepare-artifacts/

**Comment by amorde** [comment]
@nareen2 That's very strange - we could change the script to use `rm` instead of using `cat` but I don't really know why that doesn't work

We'll have a beta of 1.10 out soon

**Comment by amorde** [comment]
Going to close as the beta was released - to install the beta `[sudo] gem install cocoapods --pre`

will be included in the final release of 1.10

**Comment by Aure77** [comment]
I still having this issue with 1.10.1... Please reopen this issue.

**Comment by amorde** [comment]
@Aure77 can you open a new issue with repel steps? The bug described in this issue was fixed and released

**Comment by andreterron** [comment]
For anyone that runs into this issue, here is a workaround I found: https://sugeul.github.io/2020-06-02-xcodebuild-hang-on-CP-Prepare-artifacts/

TL;DR:
* Go to file `Pods-{app name}-artifacts.sh`
* Replace line `cat > $ARTIFACT_LIST_FILE` with `touch $ARTIFACT_LIST_FILE`
